src/Console/Component/Terminal.tsx

```diff
@@ -0,0 +1,69 @@
+import { useTerminal } from '@/utils';
+import style from './Terminal.module.css';
+import { useRef } from 'react';
+
+type TerminalValue = {
+  key: string;
+  value: string;
+};
+
+type UseTerminalReturn = {
+  currentInput: string;
+  CurrentPath: string;
+  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleKeydown: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  terminalValues: TerminalValue[];
+};
+
+type TerminalProps = {
+  path: string;
+};
```

Comment:
@Shubhamabd2002 should be optional , tabhi use hoga jab user file explorer se open in integrated terminal pe click karke terminal kholna chahega else fetch root from useDirectory hook